### PR TITLE
[lastspoke] #569 and utf8

### PR DESCRIPTION
**Pull request by bw1**

### CODE REVIEW

**Comment by ailin-nemui** [comment]
I'm worried because 

- not every channel uses utf8

**Comment by bw1** [comment]
done

File: scripts/lastspoke.pl [hunk 1]
```diff
@@ -20,11 +20,14 @@
 # Triggers on !lastspoke <nick>, !seen <nick> and !lastseen <nick>
 # 
 use strict;
+use utf8;
+use Encode qw/decode encode/;
 use Irssi;
 use Irssi::Irc;
+use JSON::PP;
```
Comment (laama1) [inline scripts/lastspoke.pl]: Requires installing of extra module perl-JSON-PP, at least on my fedora 31 server system. I installed it and am now testing this script.

**Comment by ailin-nemui** [comment]
@sndrsmnk

**Comment by sndrsmnk** [comment]
Ghe. I'm not using this script myself, anymore. Cool to see it live on. ;)

I would personally not use JSON::PP, it requires 'external' modules on almost all systems and we're not dealing with large data structures where. Some Data::Dumper & eval (evil?:P) trickery would suffice:

Eg save():
```
if (open(FD, ">dump.perlObj")) {
  print FD Dumper(%lasthash);
  close(FD);
}
```

Eg load():
```
if (open(FD, "<dump.perlObj")) {
  local $/ = undef; my $blob = <FD>; close(FD);
  %lasthash = eval 'my ' . $blob;
}
```

As for the utf-8 decode/encode stuff, why bother?

Just assume everything is utf-8 instead of encoding the eventual output to whatever your terminal is set to.

**Comment by bw1** [comment]
a new version

**Pull request merged**